tests/php/Mappers/InventoryMapperTest.php

```diff
@@ -131,27 +128,38 @@ public function dataProviderForInventoryQuantityMerge()
     public function dataProviderForGetQuantityOnHand()
     {
         return [
-            [null, null, null, "null inputs should make null quanityOnHand"],
-            [null, 0, null, "null stock with zero buffer should make null quanityOnHand"],
-            [-5, -2, -1, "stock less than negative one with invalid buffer should make negative 1 quanityOnHand"],
-            [-5, 0, -1, "stock less than negative one with zero buffer should make negative 1 quanityOnHand"],
-            [-5, 5, -1, "stock less than negative one with postiive buffer should make negative 1 quanityOnHand"],
-            [-1, -2, -1, "negative one stock with invalid buffer should make negative 1 quanityOnHand"],
-            [-1, -1, -1, "negative one stock with negative one buffer should make negative 1 quantityOnHand"],
-            [-1, 0, -1, "negative one stock with zero buffer should make negative 1 quantityOnHand"],
-            [-1, 1, -1, "negative one stock with one buffer should make negative 1 quantityOnHand"],
-            [-1, 5, -1, "negative one stock with positive buffer should make negative 1 quantityOnHand"],
-            [0, null, 0, "zero stock with null buffer should make zero quanityOnHand"],
-            [0, -2, 0,  "zero stock with negative buffer should make zero quanityOnHand"],
-            [0, 0, 0,  "zero stock with zero buffer should make zero quanityOnHand"],
-            [0, 2, 0,  "zero stock with positive buffer should make zero quanityOnHand"],
-            [1, null, 1, "one stock with null buffer should make 1 quantityOnHand"],
-            [1, 0, 1, "one stock with zero buffer should make 1 quantityOnHand"],
-            [1, 1, 0, "one stock with 1 buffer should make 0 quantityOnHand"],
-            [2, 5, 0, "two stock with five buffer should make zero quantityOnHand"],
-            [5, -2, 5, "five stock with negative buffer should make five quantityOnHand"],
-            [5, 5, 0, "five stock with five buffer should make zero quantityOnHand"],
-            [5, 2, 3, "five stock with two buffer should make three quantityOnHand"]
+            [null, null, "null stock should make null quantityOnHand"],
+            [-5, -1, "stock less than negative one should make negative 1 quantityOnHand"],
+            [-1, -1, "negative one stock should make negative 1 quantityOnHand"],
+            [0, 0, "zero stock should make zero quantityOnHand"],
+            [1, 1, "one stock should make 1 quantityOnHand"],
+            [5, 5, "five stock should make five quantityOnHand"],
+        ];
+    }
+
+    public function dataProviderForApplyStockBuffer()
+    {
+        return [
+            [null, null, null, "null inputs should make null quantityOnHand"],
+            [null, 0, null, "null input with zero buffer should make null quantityOnHand"],
+            [-5, -2, -5, "input less than negative one with invalid buffer should not change output"],

```

Comment:
This -5 eventually gets readjusted as -1 before returning?

Reply:
yeah this is due to a separation of concerns, but probably looks strange. 

- The tests for `getQuantityOnHand` still exist, and prove that we normalize the values when doing the get.

- Applying the buffer is the final action and it should not make any unrelated changes. If we had a reason to set it to -5, applying a negative buffer should not make it anything other than -5.